### DISCUSSION TITLE
SolutionUserObject and SolutionFunction for part of mesh

### DISCUSSION THREAD

**Discussion by hsheldon** [idaholab/moose#18250] [Q&A General] [answered]
I have a situation where I need to read values (e.g. porosity) from an exodus file that covers only part of the simulated domain, using SolutionUserObject and SolutionFunction. Currently this results in an error because MOOSE is trying to read values from a region that doesn't exist in the exodus file. Am I missing some way of getting round this? If not it would be useful to have the option to do this, with the ability to specify a default value where none exists.

**Comment by GiudGiud** [comment, marked as answer]
Hello

happy 4th of July!

the way of getting around this is to reload the variable in a multi app with the same mesh as the previous solution, then use a transfer to move the porosity to your simulation. You can execute this multi app at INITIAL, just once

Guillaume

**Reply by WilkAndy** [reply]
Ha, clever!

**Reply by hsheldon** [reply]
Interesting. Don't suppose there's an example somewhere that would give me some idea how to do this? I haven't played with multi apps before.

**Reply by WilkAndy** [reply]
Create an input file that just reads the previous solution into an AuxVariable called phi_from_exodus.  This probably doesn't need to have any `PorousFlow` stuff in it.  Call this input file "subapp.i"

Then, in your regular input file (the one that does the PorousFlow simulation) have an AuxVariable called porosity_aux, and simply include the following at the bottom of the file:

```
[MultiApps]
  [define_porosity]
    type = TransientMultiApp
    input_files = subapp.i
    execute_on = INITIAL
  []
[]

[Transfers]
  [populate_porosity]
    type = MultiAppInterpolationTransfer
    direction = from_multiapp
    multi_app = define_porosity
    source_variable = phi_from_exodus
    variable = porosity_aux
  []
[]
```

The `MultiAppInterpolationTransfer` has some other options you might like to use (see the doco)

**Reply by GiudGiud** [reply]
There s a dedicated multi app example on the mooseframeworks doc website. (I think, it got merged 2 weeks ago iirc). Should be in the main tabs

In the subapp input file, you can add solve=false and kernel_coverage_check=false in the Problems block to reduce the number of things to provide to make it a valid input file.

**Reply by hsheldon** [reply]
I got it working with MultiAppMeshFunctionTransfer. Thanks for your help!

**Reply by hsheldon** [reply]
It is a bit of a messy solution though. It would be sooo much simpler if SolutionFunction could handle a partial mesh.
